kore/src/Kore/Step/ClaimPattern.hs

```diff
@@ -317,6 +426,14 @@ instance From OnePathRule Attribute.RuleIndex where
 instance From OnePathRule Attribute.Trusted where
     from = Attribute.trusted . attributes . getOnePathRule
 
+instance From OnePathRule (TermLike VariableName) where
```

Comment:
It's the entire claim as a term-like. This is related to the `From rule (TermLike VariableName)` constraint used when logging (we need this otherwise we wouldn't be able to print the rule in the log). I think that discussion got lost amidst everything else. The first thing that came to mind was to use a newtype over `TermLike VariableName` so that it's clearer this is the whole rule represented as a term. What do you think?

Reply:
Yes, we could have something like

```.hs
newtype AxiomPattern = AxiomPattern (TermLike VariableName)
```